Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -3,8 +3,10 @@ Copyright (c) 2021 Kyle Miller. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kyle Miller
 -/
+import Mathlib.Algebra.BigOperators.Ring.Nat
 import Mathlib.Combinatorics.SimpleGraph.Connectivity
 import Mathlib.Combinatorics.SimpleGraph.Subgraph
+import Mathlib.SetTheory.Cardinal.Finite
```

Comment:
This is a lot of new transitive imports, but this is a small specialized (leaf?) file, so it's probably fine.

Reply:
The results here build on the Decidability instances here. I would be okay splitting it off in a separate file if that's preferred.